docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -292,6 +304,45 @@ type LocalObjectTemplate struct {
   // offered by a provider.
   Ref *corev1.ObjectReference `json:"ref"`
 }
+
+// MachineHealthCheckClass defines a MachineHealthCheck for a group of Machines. 
+// Note: all fields are defaulted on object creation through the MachineHealthCheck defaulting webhook with the exception
+// of UnhealthConditions. UnhealthyConditions is defaulted during reconciliation if it has not been profided by the user.
```

Comment:
```suggestion
// of UnhealthConditions. UnhealthyConditions is defaulted during reconciliation if it has not been provided by the user.
```

Reply:
could you expand on why `UnhealthyConditions` is special and is defaulted in the controller and not in a webhook?

Reply:
I've updated the comments on the struct. The difference is that the webhook defaults all fields other than UnhealthyConditions, and it's a required field. To avoid needing a breaking change like #5871 the idea here is to default in reconciliation.

Reply:
UX is also another reason behind defaulting; this allows to get an opinionated MHC with sane defaults by specifying an empty object

Reply:
The proposal now has no special defaulting for MHC - users will be required to define UnhealthConditions in each place they define an MHC inside the ClusterClass.